.circleci/config.yml

```diff
@@ -89,8 +89,8 @@ defaults:
   - setup_prerelease_commit_hash: &setup_prerelease_commit_hash
       name: Store commit hash and prerelease
       command: |
-        if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" ]; then echo -n > prerelease.txt; else date -u +"nightly.%Y.%-m.%-d" > prerelease.txt; fi
-        echo -n "$CIRCLE_SHA1" > commit_hash.txt
+        if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" ]; then rm cmake/prerelease.lock; else date -u +"nightly.%Y.%-m.%-d" > cmake/prerelease.lock; fi
```

Comment:
I thought we are not using the contents of the file anymore. Shouldn't we set a cmake flag instead?

Reply:
Would it be better to use "git archive" instead for the release build? If we do it differently, then building from an archive might lead to a different result than the compilation of the release build on circleci.

Reply:
> I thought we are not using the contents of the file anymore. Shouldn't we set a cmake flag instead?

Yes, this should be using the flag. The flag is now the preferred mechanism.

> Would it be better to use "git archive" instead for the release build?

Sounds good to me.

But we might be better off doing it as a separate PR to let @nishant-sachdeva finally wrap this one up.
Switching to a flag is straightforward while switching to building from source might involve some extra issues with ensuring that dependencies are downloaded, etc. so it might add some additional review cycles here.